crates/wal-protocol/src/control.rs

```diff
@@ -34,7 +34,8 @@ pub struct AnnounceLeader {
 /// Readers before v1.4.0 will crash when reading this command. For v1.4.0+, the barrier defines the
 /// minimum version of restate server that can progress after this command. It also updates the FSM
 /// in case command has been trimmed.
-#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, bilrost::Message)]
+#[derive(Debug, Clone, PartialEq, Eq, bilrost::Message)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Why was this conditional required btw? Do we pull this crate in somewhere that has no serde?